### ISSUE TITLE
Backup to the AWS S3 bucket which encrypted by AWS KMS isn't working

### ISSUE THREAD

**Issue by namm2** [vmware-tanzu/helm-charts#83]
Hi, I'm trying to get Velero working on AWS environment and hitting to an issue with encrypted S3 bucket with AWS KMS.

S3 bucket was created by Terraform:
```hcl
# Create a KMS key to encrypt S3 objects
resource "aws_kms_key" "velero" {
  deletion_window_in_days = 7
}

# Create S3 bucket for Velero backups
resource "aws_s3_bucket" "velero" {
  bucket = var.bucket_name
  acl    = "private"

  versioning {
    enabled = true
  }

  server_side_encryption_configuration {
    rule {
      apply_server_side_encryption_by_default {
        kms_master_key_id = aws_kms_key.velero.arn
        sse_algorithm     = "aws:kms"
      }
    }
  }

  tags = {
    Name        = "Velero backups"
    Environment = "Test"
  }
}
```

Created an IAM user for velero with permissions that was documented in https://github.com/vmware-tanzu/velero-plugin-for-aws#option-1-set-permissions-with-an-iam-user

Then I use the latest Helm chart to deploy to k8s cluster, with my overriden values.yaml file:
```yaml
configuration:
  provider: aws
  volumeSnapshotLocation:
    name: aws
    config:
      region: us-west-1
  backupStorageLocation:
    name: aws
    region: us-west-1
    bucket: <bucket-name>
    config:
      region: eu-central-1
      kmsKeyId: "xxxxxxxx"
initContainers:
  - name: velero-plugin-for-aws
    image: velero/velero-plugin-for-aws:v1.0.1
    volumeMounts:
      - mountPath: /target
        name: plugins

schedules:
  backup-k8s:
    schedule: "0 * * * *"
```

Results:
- Snapshots working fine
- Backups and backup logs can't be uploaded to S3 with errors:

```
time="2020-03-17T10:30:29Z" level=error msg="Error uploading log file" backup=velero-backup-k8s-20200317103012 bucket=<bucket-name> error="rpc error: code = Unknown desc = error putting object backups/velero-backup-k8s-20200317103012/velero-backup-k8s-20200317103012-logs.gz: AccessDenied: Access Denied\n\tstatus code: 403, request id: 0A9F33A087DCEE51, host id: I5KJ6oHut9691lQ41sjjxB9iMVZRJoO1Y3584+oR4QepHhChXlCS9VsbOqjLpJUIVvZHp5hf01Y=" error.file="/go/src/github.com/vmware-tanzu/velero-plugin-for-aws/velero-plugin-for-aws/object_store.go:234" error.function="main.(*ObjectStore).PutObject" logSource="pkg/persistence/object_store.go:194" prefix=
time="2020-03-17T10:30:29Z" level=info msg="Backup completed" controller=backup logSource="pkg/controller/backup_controller.go:569"
time="2020-03-17T10:30:29Z" level=error msg="backup failed" controller=backup error="rpc error: code = Unknown desc = error putting object backups/velero-backup-k8s-20200317103012/velero-backup.json: AccessDenied: Access Denied\n\tstatus code: 403, request id: FB1BCE6165C781A9, host id: mPPc+k/rRr+T0M2naqNjdXZKihl/JqtgCIAwCLa3Tgzndgi32Ms92C6/JhZd0VDh1mJ26Sc9W3w=" key=velero/velero-backup-k8s-20200317103012 logSource="pkg/controller/backup_controller.go:265"
```

I tried `aws` command without KMS key ID and it worked:
```
aws s3api put-object --bucket $BUCKET_NAME --key role.yaml --body $BODY --server-side-encryption aws:kms
{
    "ETag": "\"f1b8749fe1bfec43ee90a13a3fd20556\"",
    "ServerSideEncryption": "aws:kms",
    "VersionId": ".Cm0NH6aT5M1lj48oiCz8OVPcjGmuvpJ",
    "SSEKMSKeyId": "arn:aws:kms:eu-central-1:<AccountId>:key/<kmsKeyId>"
}
```

Then I think it's related to this issue: https://github.com/vmware-tanzu/helm-charts/issues/75 and used the patch in #76 without `kmsKeyId` in my values file and it worked
```yaml
configuration:
  provider: aws
  volumeSnapshotLocation:
    name: aws
    config:
      region: us-west-1
  backupStorageLocation:
    name: aws
    region: us-west-1
    bucket: <bucket-name>
    config:
      region: eu-central-1
      serverSideEncryption: aws:kms
initContainers:
  - name: velero-plugin-for-aws
    image: velero/velero-plugin-for-aws:v1.0.1
    volumeMounts:
      - mountPath: /target
        name: plugins

schedules:
  backup-mongodb:
    schedule: "0 * * * *"

```

Velero logs:
```
time="2020-03-17T11:00:29Z" level=info msg="Backup completed" controller=backup logSource="pkg/controller/backup_controller.go:569"
```

TL;DR: If provided S3 bucket that's encrypted with AWS KMS, Velero's Helm chart with `kmsKeyId` config is not working, instead it needs `serverSideEncryption` config.

**Comment by haofeif** [comment]
Hi, I experienced the same issue:

(1) Using S3 SSE-S3 and S3 SSE-KMS with AWS Managed Key work, as it does not require providing the kmsKeyID into the `values.configuration.backupStorageLocation.config.kmsKeyID`
(2) However, using S3 SSE-KMS with CMK does not work. It is successful with backup, but cannot retrieve the logs nor doing restore.
```bash
$ velero backup create velero01 --include-namespaces velero
Backup request "velero01" submitted successfully.
Run `velero backup describe velero01` or `velero backup logs velero01` for more details.
 $ velero backup get
NAME       STATUS      ERRORS   WARNINGS   CREATED                          EXPIRES   STORAGE LOCATION   SELECTOR
velero01   Completed   0        0          2021-09-19 10:58:38 +1000 AEST   29d       default            <none>
$ velero backup logs velero01
An error occurred: request failed: <?xml version="1.0" encoding="UTF-8"?>
<Error><Code>AccessDenied</Code><Message>Access Denied</Message><RequestId>1ZNH7SDFEFWBFW0Z</RequestId><HostId>ntIpqOimFgPGEipnIqlUe4g/tT6xITLWocnmVs4sowBNTkzhw6yn/IkrQymnPfsffsadfsfsafsaf=</HostId></Error>
```

**Comment by haofeif** [comment]
Is there any update on this ? As this stop velero leveraging the [S3 Bucket Keys](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucket-key.html) feature which requires using AWS CMK (Custom Managed Key) to achieve "reduce AWS KMS request costs by up to 99 percent by decreasing the request traffic from Amazon S3 to AWS KMS."

**Comment by ditahk** [comment]
Is there any update on this?
Is velero able to encrypt data in transit?
